File: coding-club/Session 1/kazi.cpp
// https://www.codechef.com/problems/SPIT1

#include <iostream>
#include <map>

using namespace std;

int main() {
    map <string,int> m;
    int n;
    cin >> n;
    int maxVal = 0, v;
    string s, maxName;
    for (int i=1; i<=n; i++) {
        cin >> s;
        cin >> v;
        m[s] += v;
        if (m[s] > maxVal) {
            maxVal = m[s];
            maxName = s;
        }
    }
    cout << maxName;
    return 0;
}

File: coding-club/Session 1/kazi.py
# https://www.codechef.com/problems/SPIT1

n = int(input())
dic = dict()

maxVal = 0
maxName = ''

for _ in range(n):
    s,v = input().split()
    v = int(v)
    try:
        dic[s] += v
    except:
        dic[s] = v
    if dic[s] > maxVal:
        maxVal = dic[s]
        maxName = s

print(maxName)